src/plugins/query_enhancements/server/utils/facet.ts

```diff
@@ -40,7 +40,7 @@ export class Facet {
       const query: Query = request.body.query;
       const { format, df } = request.body;
       const params = {
-        body: { ...query },
+        body: { query: query.query },
```

Comment:
Is there a reason why we removed the rest of the query object? I don't remember if the `fetch()` function for `PPL` and `SQL` use the rest of the object off the top of my head.

Reply:
Before this change the request body that we used to send was of the following format
```
{
    query: {
      query: 'SELECT * FROM aaa* LIMIT 10',
      language: 'SQL',
      dataset: [Object],
      format: 'jdbc'
    },
    aggConfig: null,
    df: null
}
```
However the SQL plugins request only needs the following parameters.
![image](https://github.com/user-attachments/assets/94ca4827-c663-4b67-966a-1be40fd3dd51)

Also we observed sending in the extra unrecognized parameters was causing bugs where the SQL Search not working for IndexPatterns with Wildcards in the name, therefore this PR aims to cleanup the request body that we send.

Reply:
yeah end up doing something similar here too https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7943/files